File: checkio_js/home/javascript/allTheSame.js
import assert from "assert";

function allTheSame(elements) {
    return elements.every(a => a===elements[0])
}


console.log('Example:')
console.log(allTheSame([1, 1, 1]))

// These "asserts" are used for self-checking and not for an auto-testing

assert.equal(allTheSame([1, 1, 1]), true)
assert.equal(allTheSame([1, 2, 1]), false)
assert.equal(allTheSame(['a', 'a', 'a']), true)
assert.equal(allTheSame([]), true)
assert.equal(allTheSame([1]), true)
console.log("Coding complete? Click 'Check' to earn cool rewards!");

File: checkio_js/home/python/allTheSame.py


def allTheSame(elements):
    return all([x == elements[0] for x in elements])


if __name__ == '__main__':
    print('Example:')
    print(allTheSame([1, 1, 1]))

# These "asserts" are used for self-checking and not for an auto-testing

assert allTheSame([1, 1, 1]) == True
assert allTheSame([1, 2, 1]) == False
assert allTheSame(['a', 'a', 'a']) == True
assert allTheSame([]) == True
assert allTheSame([1]) == True
print("Coding complete? Click 'Check' to earn cool rewards!")
